internal/generator/vector/output/elasticsearch/elasticsearch.go

```diff
@@ -48,203 +43,55 @@ api_version = "v{{ .Version }}"
-      }
-    }
-  }
-}`
-)
-
 func (e *Elasticsearch) SetCompression(algo string) {
 	e.Compression.Value = algo
 }
 
-func New(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret, strategy common.ConfigStrategy, op Options) []Element {
-	outputs := []Element{}
-
-	esIndexID := helpers.MakeID(id, "add_es_index")
-	dedotID := helpers.MakeID(id, "dedot_and_flatten")
-
+func New(id string, o obs.OutputSpec, inputs []string, secrets helpers.Secrets, strategy common.ConfigStrategy, op Options) []Element {
 	if genhelper.IsDebugOutput(op) {
 		return []Element{
-			SetESIndex(esIndexID, inputs, o, op),
-			FlattenLabels(dedotID, []string{esIndexID}),
-			Debug(id, helpers.MakeInputs([]string{dedotID}...)),
+			Debug(id, helpers.MakeInputs(inputs...)),
 		}
 	}
-	sink := Output(id, o, []string{dedotID}, secret, op)
+	sink := Output(id, o, inputs, secrets, op)
 	if strategy != nil {
 		strategy.VisitSink(sink)
 	}
-	request := common.NewRequest(id, strategy)
-	request.TimeoutSecs.Value = 2147483648
-	outputs = MergeElements(outputs,
-		[]Element{
-			SetESIndex(esIndexID, inputs, o, op),
-			FlattenLabels(dedotID, []string{esIndexID}),
-			sink,
-			common.NewAcknowledgments(id, strategy),
-			common.NewBatch(id, strategy),
-			common.NewBuffer(id, strategy),
-			request,
-		},
-
-		common.TLS(id, o, secret, op),
-		BasicAuth(id, o, secret),
-	)
+	outputs := []Element{
+		sink,
+		common.NewAcknowledgments(id, strategy),
+		common.NewBatch(id, strategy),
+		common.NewBuffer(id, strategy),
+		common.NewRequest(id, strategy),
+		tls.New(id, o.TLS, secrets, op),
+		BasicAuth(id, o, secrets),
+	}
 
 	return outputs
 }
 
-func Output(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret, op Options) *Elasticsearch {
+func Output(id string, o obs.OutputSpec, inputs []string, secrets helpers.Secrets, op Options) *Elasticsearch {
 	es := Elasticsearch{
 		ComponentID: id,
-		Endpoint:    o.URL,
+		Endpoint:    o.Elasticsearch.URL,
 		Inputs:      helpers.MakeInputs(inputs...),
+		Index:       o.Elasticsearch.Index,
 		RootMixin:   common.NewRootMixin(nil),
-	}
-	// If valid version is specified
-	if o.Elasticsearch != nil && o.Elasticsearch.Version > 0 {
-		es.Version = o.Elasticsearch.Version
-	} else {
-		es.Version = logging.DefaultESVersion
+		Version:     o.Elasticsearch.Version,
 	}
 	return &es
 }
 
-func BasicAuth(id string, o logging.OutputSpec, secret *corev1.Secret) []Element {
-
-	conf := []Element{}
-
-	if o.Secret != nil {
-		hasBasicAuth := false
-		conf = append(conf, BasicAuthConf{
-			Desc:        "Basic Auth Config",
-			ComponentID: id,
-		})
-		if common.HasUsernamePassword(secret) {
-			hasBasicAuth = true
-			up := UserNamePass{
-				Username: common.GetFromSecret(secret, constants.ClientUsername),
-				Password: common.GetFromSecret(secret, constants.ClientPassword),
-			}
-			conf = append(conf, up)
+func BasicAuth(id string, o obs.OutputSpec, secrets helpers.Secrets) Element {
```

Comment:
You refactored `HTTPAuth` into `common/auth/auth.go`, doesn't ` HTTPAuth()`  in that file do this? Shouldn't you use that instead of refactoring this method?